src/AutoRest.CSharp/Output/Models/Types/ExtensionDetection.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+using System;
+using System.Collections.Generic;
+using AutoRest.CSharp.Output.Models.Requests;
+using AutoRest.CSharp.Input;
+
+namespace AutoRest.CSharp.Output.Models.Type.Decorate
+{
+    internal static class ExtensionDetection
+    {
+        public static bool IsExtension(OperationGroup operationGroup)
```

Comment:
Having something that is an extension for the OperationGroup and is later written into a property on the  OperationGroup is double confusing.

Reply:
Not sure I follow, this isn't an extension method on an operationgroup it is a method which determines whether an operationgroup is an extension resource type in ARM and then sets the result of this as a property on the operationgroup

Reply:
My bad, was under the impression this was an extension method.